File: PDFill/Models/FormFill.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace PDFill.Models
{
    public class FormFill
    {
        public List<Item> Items { get; set; } = new List<Item>();
        public string Filename { get; set; }
        public DocFile DocFile { get; set; }
        
    }
    public class Item
    {
        public string Key { get; set; }
        public string Normalized { get; set; }
        public string Value { get; set; }
    }
    public enum DocFile
    {
        Doc,
        Docx,
        XML
    }
}

File: PDFill/ClientApp/src/app/models/FormFill.ts
export interface IFormFill {
    filename: string;
    items: Item[];
    docFile: DocFile;
}
export class FormFill {
    filename: string = "";
    item: Item[] = [];
    docfile: DocFile = DocFile.Docx;
}
export class Item {
    key: string = "";
    normalized: string = "";
    value: string = "";
}
export enum DocFile {
    Doc,
    Docx,
    XML
}